components/AIEstimation.tsx

```diff
@@ -0,0 +1,60 @@
+// components/ai-estimation.tsx
+'use client'
+
+import { useState } from 'react'
+import { useMutation } from 'convex/react'
+import { api } from '@/convex/_generated/api'
+import { Button } from "@/components/ui/button"
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
+import { Spinner } from "@/components/ui/spinner"
+import { useToast } from '@/hooks/use-toast'
+
+interface AIEstimationProps {
+    vehicleDetails: any // Replace with proper type
+    selectedServices: string[]
+    customizations: any // Replace with proper type

```

Comment:
**Suggestion:** Replace the `any` type with specific types for `vehicleDetails` and `customizations` to ensure type safety and better maintainability. [best practice, importance: 9]
```suggestion
interface AIEstimationProps {
    vehicleDetails: VehicleDetailsType // Specific type
    customizations: CustomizationsType // Specific type
}
```